File: Number_of_Times_Binary_String_Is_Prefix_Aligned.java
class Solution {
    public int numTimesAllBlue(int[] flips) {
        long sum=0;
        long expected=0;
        int j=1;
        int count=0;
        for (int i:flips)
        {
            sum+=i;
            expected+=j++;
            if(sum==expected){count++;}
        }
        return count;
    }
}

File: Number_of_Times_Binary_String_Is_Prefix_Aligned.cpp
class Solution {
public:
    int numTimesAllBlue(vector<int>& l) {
        int n = l.size();
        vector<int> num(n , 0);
        for(int i = 0 ; i < n ; i++){
            num[l[i]-1] = i+1;
        }
        int day =1; 
        int moments = 0;
        for(int i = 0 ; i < n ; i++){
            day = max(day , num[i]);
            if(i+1 == day)moments++;
        }
        return moments;
    }
};
